projects/kit/components/items-with-more/items-with-more.directive.ts

```diff
@@ -1,24 +1,45 @@
 import type {OnChanges} from '@angular/core';
 import {Directive, Input} from '@angular/core';
+import {tuiInjectElement} from '@taiga-ui/cdk/utils/dom';
 import {Subject} from 'rxjs';
 
 @Directive({
     standalone: true,
+    host: {
+        '[class._multiline]': 'linesLimit > 1',
+        '[style.--t-min-width.px]': 'maxWidth()',
+    },
 })
 export class TuiItemsWithMoreDirective implements OnChanges {
+    private readonly el = tuiInjectElement();
+    private moreSide: 'end' | 'start' = 'end';
+
     @Input()
     public itemsLimit = Infinity;
 
     @Input()
     public required = -1;
 
     @Input()
-    public side: 'end' | 'start' = 'end';
+    public linesLimit = 1;
 
     // TODO: refactor to signal inputs after Angular update
     public readonly change$ = new Subject<void>();
 
+    @Input('side')
+    public set setSide(side: 'end' | 'start') {
+        this.moreSide = side;
+    }
+
+    public get side(): 'end' | 'start' {
+        return this.linesLimit > 1 ? 'end' : this.moreSide;
+    }
+
     public ngOnChanges(): void {
         this.change$.next();
     }
+
+    protected maxWidth(): number {
```

Comment:
Would this cause some unexpected spaces if there's a huge item that is way down the list and is hidden?